### DISCUSSION TITLE
Set up OHIF plugin with local datastore

### DISCUSSION THREAD

**Discussion by kunjesh1** [Project-MONAI/MONAILabel#1215] [General]
Hi Team, 

I am planning to set up OHIF viewer using MONAI Label for our AI workflows. But, the bottleneck is we don't use dicomweb so just wanted to know :

1. How to start the MONAI label server with the local datastore(file archive) and use the upload image API for uploading the image. 
2. Is it also possible to use the s3 bucket link where our images are stored

Right now I have used this command to start the MONAI label server but the upload image API is greyed out

```monailabel start_server --app apps/radiology --studies datasets --conf models segmentation_spleen```

Getting this error, upload image is greyed out

<img width="1364" alt="Screenshot 2022-12-16 at 3 02 47 PM" src="https://user-images.githubusercontent.com/16972191/208068149-ac3bc3ea-b8b5-41f8-b25d-7a1a550d4471.png">

**Comment by lassoan** [comment]
Some groups use a shared MONAILabel server and each user connects to it via 3D Slicer. The only extra step for users is that they need to install 3D Slicer on their computer. (You can run Slicer instances on the server in docker containers and then users can connect simply in their web browser; but this is much more work to set up on the server, so it is not commonly done.)

**Reply by kunjesh1** [reply]
Thanks, @lassoan for the quick reply. Without slicer not possible? I have done some research and people have written different implementations of datastore for xnat and flywheel database but don't know how it works, if I am able to set up the local file archive also and use the /upload API my problem is solved

Flywheel datastore
https://github.com/Project-MONAI/MONAILabel/discussions/1124
 
Xnat data store 
https://github.com/Project-MONAI/MONAILabel/blob/main/monailabel/datastore/xnat.py#L92
 
Local data store code 
 https://github.com/Project-MONAI/MONAILabel/blob/main/monailabel/datastore/local.py

**Reply by lassoan** [reply]
You can use many different datastores in MONAILabel. However, if you use OHIF as a client then your only option is DICOMweb. If you use 3D Slicer as a client then there is no limitation on what datastore you use.

**Reply by kunjesh1** [reply]
Oh Okay, just wanted to know how to set up the datastore part, since I am using OHIF v3 with cornerstone 2D as an extension which supports Dicom JSON also,so I have customized that according to my use case
Is there any command for spinning up the MONAI server with the local file archive,I have not found it in the documentation

**Comment by SachidanandAlle** [comment]
The grey one are deprecated over other ones.. u have PUT image API but unless the data store is not supporting.. it's not much of use.. example.. file system supports but not xnat..

**Reply by kunjesh1** [reply]
Thanks :), Is there any command for spinning up the MONAI server with the custom data store, I have not found it in the documentation

**Reply by SachidanandAlle** [reply]
They r not pluggable via command.. you can initial your data store in your app.. simply derive init data store method in your main.py and return the same

**Reply by kunjesh1** [reply]
Thanks, let me try that :)
